File: fetch_buildings.py
import urllib.request
import json

query = """
[out:json];
(
  way["building"](49.180,-123.200,49.205,-123.155);
);
out geom;
"""

req = urllib.request.Request(
    'https://overpass-api.de/api/interpreter',
    data=query.encode('utf-8'),
    headers={'User-Agent': 'Mozilla/5.0'}
)

try:
    with urllib.request.urlopen(req) as response:
        data = json.loads(response.read().decode('utf-8'))
        buildings = []
        for el in data.get('elements', []):
            if 'geometry' in el:
                buildings.append([[g['lat'], g['lon']] for g in el['geometry']])
        
        with open('public/yvr_buildings.json', 'w') as f:
            json.dump({'buildings': buildings}, f)
        print(f"Saved {len(buildings)} buildings.")
except Exception as e:
    print(e)

File: tools/fetch_buildings.js
const https = require('https');
const fs = require('fs');

const query = `
[out:json];
(
  way["aeroway"="terminal"](49.18,-123.20,49.20,-123.16);
  way["building"](49.18,-123.20,49.20,-123.16);
);
out geom;
`;

const req = https.request({
  hostname: 'overpass-api.de',
  path: '/api/interpreter',
  method: 'POST',
  headers: {
    'Content-Type': 'application/x-www-form-urlencoded'
  }
}, (res) => {
  let data = '';
  res.on('data', chunk => data += chunk);
  res.on('end', () => {
    const parsed = JSON.parse(data);
    const buildings = parsed.elements.map(el => {
      if (!el.geometry) return null;
      return el.geometry.map(g => [g.lat, g.lon]);
    }).filter(Boolean);
    
    fs.writeFileSync('public/yvr_buildings.json', JSON.stringify({ buildings }));
    console.log(`Saved ${buildings.length} buildings.`);
  });
});
req.write(query);
req.end();
